library/include/LuaWrapper.h

```diff
@@ -127,6 +127,8 @@ namespace LuaWrapper {
     struct DFRefHeader {
         void *ptr;
         const struct_field_info *field_info;
+        const void *tag_ptr;
+        const type_identity *tag_identity;
```

Comment:
Is this something that could go in `struct_field_info` instead? Or is this a "back-reference" from the field being tagged to the field acting as the tag? Since (as far as I know) DFRefHeader objects are created every time a DF object/field is accessed from Lua, I'd like to keep the size as small as possible (but I suppose this might not have a significant impact if they are garbage-collected).

Reply:
If you're accessing a vector item, I'm pretty sure only the first field of DfRefHeader is currently non-null. The two added fields are only used (in code that has yet to be written) in three cases: when a tagged union is accessed from a struct, when a vector of tagged unions is accessed from a struct, and when an item of that vector is accessed.

If there's some other way to store the information (like how type identities are stored), I'm not familiar enough with lua to know how it's done, but it seems like that could be a more memory-efficient way to handle it.